be/src/runtime/lake_tablets_channel.cpp

```diff
@@ -781,6 +793,87 @@ Status LakeTabletsChannel::incremental_open(const PTabletWriterOpenRequest& para
     return Status::OK();
 }
 
+void LakeTabletsChannel::update_profile() {
+    if (_profile == nullptr) {
+        return;
+    }
+
+    bool expect = false;
+    if (!_is_updating_profile.compare_exchange_strong(expect, true)) {
+        // skip concurrent update
+        return;
+    }
+    DeferOp defer([this]() { _is_updating_profile.store(false); });
+    _profile->inc_version();
+    COUNTER_UPDATE(_profile_update_counter, 1);
+    SCOPED_TIMER(_profile_update_timer);
+
+    std::vector<AsyncDeltaWriter*> async_writers;
+    {
+        std::shared_lock<bthreads::BThreadSharedMutex> lk(_rw_mtx);
+        async_writers.reserve(_delta_writers.size());
+        for (auto& [tablet_id, delta_writer] : _delta_writers) {
+            async_writers.push_back(delta_writer.get());
+        }
+    }
+
+    std::vector<RuntimeProfile*> tablets_profile;
+    for (auto async_writer : async_writers) {
+        DeltaWriter* writer = async_writer->delta_writer();
+        RuntimeProfile* profile = _tablets_profile->create_child(fmt::format("{}", writer->tablet_id()));
+        _update_tablet_profile(writer, profile);
+    }
+
+    ObjectPool obj_pool;
+    if (!tablets_profile.empty()) {
+        auto* merged_profile = RuntimeProfile::merge_isomorphic_profiles(&obj_pool, tablets_profile);
+        RuntimeProfile* final_profile = _profile->create_child("PeerReplicas");
+        auto* tablets_counter = ADD_COUNTER(final_profile, "TabletsNum", TUnit::UNIT);
+        COUNTER_UPDATE(tablets_counter, tablets_profile.size());
+        final_profile->copy_all_info_strings_from(merged_profile);
+        final_profile->copy_all_counters_from(merged_profile);
+    }
+}
+
+#define ADD_AND_UPDATE_COUNTER(profile, name, type, val) (ADD_COUNTER(profile, name, type))->update(val)
+#define ADD_AND_UPDATE_TIMER(profile, name, val) (ADD_TIMER(profile, name))->update(val)
+
+void LakeTabletsChannel::_update_tablet_profile(DeltaWriter* writer, RuntimeProfile* profile) {
+    const lake::DeltaWriterStat& writer_stat = writer->get_writer_stat();
+    ADD_AND_UPDATE_COUNTER(profile, "WriterTaskCount", TUnit::UNIT, writer_stat.task_count);
+    ADD_AND_UPDATE_TIMER(profile, "WriterTaskPendingTime", writer_stat.pending_time_ns);
+    ADD_AND_UPDATE_COUNTER(profile, "WriteCount", TUnit::UNIT, writer_stat.write_count);
+    ADD_AND_UPDATE_COUNTER(profile, "RowCount", TUnit::UNIT, writer_stat.row_count);
+    ADD_AND_UPDATE_TIMER(profile, "WriteTime", writer_stat.write_time_ns);
+    ADD_AND_UPDATE_COUNTER(profile, "MemtableFullCount", TUnit::UNIT, writer_stat.memtable_full_count);
+    ADD_AND_UPDATE_COUNTER(profile, "MemoryExceedCount", TUnit::UNIT, writer_stat.memory_exceed_count);
+    ADD_AND_UPDATE_TIMER(profile, "WriteWaitFlushTime", writer_stat.write_wait_flush_tims_ns);
+    ADD_AND_UPDATE_TIMER(profile, "CloseTime", writer_stat.write_wait_flush_tims_ns);
+    ADD_AND_UPDATE_TIMER(profile, "FinishTime", writer_stat.finish_time_ns);
```

Comment:
it's better to add a threshold for these counters, otherwise they can be pretty noisy